### PR TITLE
add APIs for dash docs testings

### PR DESCRIPTION
**Pull request by byronz**
*Start with a description of this PR. Then edit the list below to the items that make sense for your PR scope, and check off the boxes as you go!*

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] add timeout argument in process runner
   -  [ ] add wait_for_element_by_id
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))

### optionals

- [x] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follow
    -  [ ] this github [#PR number]() updates the dash docs

### CODE REVIEW

File: dash/testing/browser.py [hunk 1]
```diff
@@ -173,6 +173,20 @@ def wait_for_element_by_css_selector(self, selector, timeout=None):
             ),
         )
 
+    def wait_for_element_by_id(self, element_id, timeout=None):
```
Comment (alexcjohnson) [inline dash/testing/browser.py]: Is it really worth adding this, vs `wait_for_element_by_css_selector('#mi-id')`?

Comment (byronz) [reply dash/testing/browser.py]: Yes the id we have like `#wait-/testing` is an invalid CSS selector

Comment (alexcjohnson) [reply dash/testing/browser.py]: hmm OK - seems like poor form to use a character in an ID that can't be used in CSS, but I guess sometimes we can't control that.

File: dash/testing/application_runners.py [hunk 2]
```diff
@@ -179,23 +179,34 @@ def __init__(self, keep_open=False, stop_timeout=3):
         self.proc = None
 
     # pylint: disable=arguments-differ
-    def start(self, app_module, application_name="app", port=8050):
+    def start(
```
Comment (byronz) [inline dash/testing/application_runners.py]: @alexcjohnson  played a little bit on the dash-docs side

this is somehow a little break change, but I guess it's not used by anyone in the community.  the raw_command option would give us more flexibility in the future.  like in dash-docs case, our usage should just use python run.py instead a gunicorn or waitress which targets for production.

File: dash/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 ## Unreleased
 
+### Added
+
+- [#918](https://github.com/plotly/dash/pull/918) Adds `wait_for_element_by_id` and `visit_and_snapshot` APIs in browser, adds `raw_command` option (it aslo has higher priority than
```
Comment (alexcjohnson) [inline dash/CHANGELOG.md]:
```suggestion
- [#918](https://github.com/plotly/dash/pull/918) Adds `wait_for_element_by_id` and `visit_and_snapshot` APIs in browser, adds `raw_command` option (it also has higher priority than
```

**Pull request merged**